interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -287,10 +298,24 @@ public void tearDown() throws Exception {
     if (channel != null) {
       channel.shutdown();
     }
+    stopServer();
   }
 
   protected abstract ManagedChannel createChannel();
 
+  /**
+   * Returns the server builder used to create server for each test run.  Return {@code null} if
+   * it shouldn't start a server in the same process.
+   */
+  @Nullable
+  protected AbstractServerImplBuilder<?> getServerBuilder() {
+    return null;
+  }
+
+  protected List<? extends ServerInterceptor> getServerInterceptors() {
```

Comment:
Could we just let subclasses call `serverBuilder.intercept(...)` before returning in `getServerBuilder()`?

Reply:
wouldn't that mean sharing the server between this class and the subclasses?

Reply:
Done.

Reply:
@carl-mastrangelo, I don't know what you mean. Kun made the change, so you can point out an issue with it if you see one.